apps/emqx/include/emqx_session_mem.hrl

```diff
@@ -40,14 +40,14 @@
     %% Next packet id of the session
     next_pkt_id = 1 :: emqx_types:packet_id(),
     %% Retry interval for redelivering QoS1/2 messages (Unit: millisecond)
-    retry_interval :: timeout(),
+    retry_interval = 0 :: timeout(),
```

Comment:
Does 0 have a special meaning for session_mem (disable retry)? Otherwise it looks pretty scary.

Reply:
It has, yes.
https://github.com/emqx/emqx/blob/923898eadf0a3a67668217cc751f2633138fa5d6/apps/emqx/src/emqx_channel.erl#L1366-L1367

These changes are actually only needed to please dialyzer, but given how many questions they raise, I'll try to please him another way.